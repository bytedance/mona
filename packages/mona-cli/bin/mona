#! /usr/bin/env node

const mona = require('../dist').default;
mona();
