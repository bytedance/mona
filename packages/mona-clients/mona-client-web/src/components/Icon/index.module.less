.container {
  display: inline-block;
  font-size: 0;

  .icon {
    font: normal normal normal 14px/1 "ttui";

    &::before {
      margin: 0;
      box-sizing: border-box;
    }

    &.success {
      color: #F95959;
    }

    &.success:before {
      content: "\e613";
    }

    &.success_no_circle {
      color: #F95959;
    }

    &.success_no_circle:before {
      content: "\e610";
    }

    &.info {
      color: #CACACA;
    }

    &.info:before {
      content: "\e60d";
    }

    &.warn {
      color: #F5A623;
    }

    &.warn:before {
      content: "\e614";
    }

    &.waiting {
      color: #50ABF9;
    }

    &.waiting:before {
      content: "\e612";
    }

    &.clear {
      color: #F95959;
    }

    &.clear:before {
      content: "\e615";
    }

    &.cancel {
      color: #222222;
    }

    &.cancel:before {
      content: "\e611";
    }

    &.download {
      color: #222222;
    }

    &.download:before {
      content: "\e60f";
    }

    &.search {
      color: #222222;
    }

    &.search:before {
      content: "\e60e";
    }
  }
}