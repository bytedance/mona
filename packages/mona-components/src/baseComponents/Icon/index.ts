import React from 'react';
import createComponent from '../../createComponent';
import { BaseProps } from '../base';

export interface IconProps extends BaseProps {
  type: 'success' | 'success_no_circle' | 'info' | 'warn' | 'warning' | 'clear' | 'cancel' | 'download' | 'search'
  size?: number;
  color?: string;
}

export const Icon: React.ComponentType<IconProps> = createComponent<IconProps>('icon')
