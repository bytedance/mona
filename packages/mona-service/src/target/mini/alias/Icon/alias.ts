import { bubbleEventsAlias, basePropAlias, propAliasMap } from '../prop';
import { IconAlias as Alias } from '../type';

const alias: Alias = {
  ...basePropAlias,
  ...bubbleEventsAlias,
  size: propAliasMap.size,
  color: propAliasMap.loading,
  type: propAliasMap.type,
};

export default alias;
