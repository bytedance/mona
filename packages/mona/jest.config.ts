import baseConfig from '../../jestconfig.base';

module.exports = baseConfig;
