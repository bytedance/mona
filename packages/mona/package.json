{
  "name": "@bytedance/mona",
  "version": "0.2.0-alpha.9",
  "description": "Merchant app's developing and building tools",
  "main": "./dist/index",
  "scripts": {
    "build": "yarn clean && ttsc -b",
    "start": "ttsc -b -w",
    "clean": "rm -rf dist",
    "test": "jest"
  },
  "files": [
    "dist"
  ],
  "author": "zhangxiaowei.chris@bytedance.com",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "devDependencies": {
    "copy-webpack-plugin": "^10.2.0",
    "react": "^17.0.2",
    "webpack": "^5.58.2",
    "webpack-chain": "^6.5.1"
  },
  "peerDependencies": {
    "react": "^17.0.2"
  },
  "gitHead": "cc5aa5283af595fd7d8a7b4921f54e2797504edd"
}
