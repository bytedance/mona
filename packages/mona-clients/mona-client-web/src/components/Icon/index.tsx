import React from 'react';
import { IconProps } from '@bytedance/mona';
import styles from './index.module.less';
import { useHandlers } from '../hooks';

const Icon: React.FC<IconProps> = props => {
  const { children, type, size = 24, color, ...restProps } = props;
  const { handleClassName, ...handleProps } = useHandlers(restProps);

  return (
    <div className={handleClassName(styles.container)} {...handleProps}>
      <i className={`${styles.icon} ${!!type ? styles[type] : ''}`} style={{ color, fontSize: size }}></i>
    </div>
  );
};

export default Icon;
