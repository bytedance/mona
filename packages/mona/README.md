# @bytedance/mona
对外暴露提供构建时API
## 配置创建
### createProjectConfig
创建项目配置，`mona.config.ts`中使用
#### 例子
```js
// mona.config.ts
import { createProjectConfig } from '@bytedance/mona';
export default createProjectConfig({
  projectName: 'demo',
  input: './src/app.tsx',
  output: 'dist'
})
```

#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| projectName  | 应用/插件名称 | 是 | string | - |
| input  | 入口文件路径 | 否 | string | - |
| output  | 打包后的文件目录路径 | 否 | string | - |
| raw  | 定义函数来修改webpack配置 | 否 | (options: any) => any | - |
| dev  | 用来修改本地服务配置 | 否 | string | { port: 9999 } |

### createAppConfig
创建应用配置，`app.config.ts`中使用
#### 例子
```js
// app.config.ts
import { createAppConfig } from '@bytedance/mona';
export default createAppConfig({
  pages: [
    'pages/Home/index',
    'pages/Info/index',
    'pages/List/index'
  ]
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| pages  | 页面路径 | 是 | string[] | - |

默认`pages`的第一项为首页
### createPageConfig
创建页面配置，需在每个页面目录的`page.config.ts`中使用

#### 例子
```js
// page.config.ts
import { createPageConfig } from '@bytedance/mona';
export default createPageConfig({
  navigationBarTitleText: 'page title'
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| navigationBarTitleText  | 当前页面标题 | 否 | string | - |
