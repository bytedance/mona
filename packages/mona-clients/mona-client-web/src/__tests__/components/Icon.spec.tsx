import React from 'react';
import { mount } from 'enzyme';
import Icon from '../../components/Icon';
import mountTest from '../../../../../tests/shared/mountTest';
import handlerTest from '../../../../../tests/shared/handlerTest';
import { IconProps } from '@bytedance/mona';

describe('web component: Icon', () => {
  mountTest(Icon)
  handlerTest({ Component: Icon });

  it('should render correctly', () => {
    expect(mount(<Icon type="success" />).render()).toMatchSnapshot();
  })
  
  it('should have right properties', () => {
    const props: IconProps = {
      type: 'success',
      size: 20,
      color: 'red',
      className: 'class-name',
      style: { color: 'red' },
    }
    const wrapper = mount(<Icon {...props} />);
    expect(wrapper.render()).toMatchSnapshot();
  })
})
