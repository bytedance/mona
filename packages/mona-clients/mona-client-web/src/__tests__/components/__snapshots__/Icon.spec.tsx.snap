// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web component: Icon should have right properties 1`] = `
LoadedCheerio {
  "0": <div
    class="container class-name  "
    style="color: red;"
  >
    <i
      class="icon success"
      style="color: red; font-size: 20px;"
    />
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`web component: Icon should render correctly 1`] = `
LoadedCheerio {
  "0": <div
    class="container   "
  >
    <i
      class="icon success"
      style="font-size: 24px;"
    />
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
